lib/foreman/logging.rb

```diff
@@ -119,27 +136,37 @@ def build_console_appender
       return unless @config[:console_inline]
 
       ::Logging.logger.root.add_appenders(
-        ::Logging.appenders.stdout(:layout => build_layout(@config[:pattern], @config[:colorize]))
+        ::Logging.appenders.stdout(:layout => build_layout)
       )
     end
 
-    # currently we support two types of appenders, rolling file and syslog
-    # note that syslog ignores pattern and logs only messages
     def build_root_appender(options)
       name = "foreman"
 
       case @config[:type]
       when 'syslog'
         build_syslog_appender(name, options)
+      when 'journal', 'journald'
+        build_journald_appender(name, options)
       when 'file'
         build_file_appender(name, options)
       else
         fail 'unsupported logger type, please choose syslog or file'
       end
     end
 
+    def options_with_syslog_facility(options, defaults)
+      opts = options.reverse_merge(defaults)
+      opts[:facility] = self.class.const_get("::Syslog::Constants::#{opts[:facility]}")
```

Comment:
As a developer, in which case I would want to override the `:facility`? Should generic foreman code expose this via API?

Reply:
User wants to change facility when processing logs in syslog. Not following.

Reply:
I take it as "also log foreman exception code along with other fields", good idea. Added:

    extra_fields[:foreman_code] = exception.code if exception.respond_to?(:code)

Reply:
I haven't meant that (I mean `foreman source code`), but good idea anyway :)

I might just not understand the meaning of the `:facility` and when in the **source code** would I want to use the non-default one.